### PR TITLE
Setup GH Actions

### PR DESCRIPTION
**Pull request by xbill82**
Introduce GH Actions, using the new cross-repo doc deploy system.

### CODE REVIEW

**Comment by alexandrebouthinon** [comment]
Why not just merge 2-dev into master? By duplicating the changes in two different commits, it may lead to conflicts in a future release 😅

**Comment by xbill82** [comment]
@alexandrebouthinon bc `2-dev` has other differences from `master` and we are not releasing yet (or, at least, setting up a CI shouldn't imply a release). Plus, the files are exactly the same, this commit shouldn't introduce any conflicts.

File: .github/workflows/pull_request.workflow.yml [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup dotnet
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: ${{ matrix.dotnet }}
+      - uses: ./.github/actions/unit-tests
+
+  snippets-tests:
+    name: Snippet Tests - .NET v${{ matrix.dotnet }}
```
Comment (alexandrebouthinon) [inline .github/workflows/pull_request.workflow.yml]:
```suggestion
    name: Snippet Tests
```

File: .github/workflows/push_dev.workflow.yml [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup dotnet
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: ${{ matrix.dotnet }}
+      - uses: ./.github/actions/unit-tests
+
+  snippets-tests:
+    name: Snippet Tests - .NET v${{ matrix.dotnet }}
```
Comment (alexandrebouthinon) [inline .github/workflows/push_dev.workflow.yml]:
```suggestion
    name: Snippet Tests
```

File: .github/workflows/push_master.workflow.yml [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+      - uses: webfactory/ssh-agent@v0.5.2
+        with:
+          ssh-private-key: ${{ secrets.SSH_KEY }}
+      - uses: ./.github/actions/dead-links
+
+  deploy-docs-prod:
+    name: Documentation deployment
+    runs-on: ubuntu-18.04
+    timeout-minutes: 30
+    needs: [dead-links, deploy-prod]
```
Comment (alexandrebouthinon) [inline .github/workflows/push_master.workflow.yml]: Dead link check should be run before NuGet deployment

**Review by alexandrebouthinon — CHANGES_REQUESTED** [review]

**Comment by xbill82** [comment]
Closing as feature is provided in #70, which will be merged to `master`

**Pull request closed without merging**